metastore/metastore-api/src/main/java/org/apache/drill/metastore/util/TableMetadataUtils.java

```diff
@@ -46,21 +46,24 @@
    * @param type type of the column
    * @return {@link Comparator} instance
    */
-  public static Comparator getComparator(TypeProtos.MinorType type) {
+  @SuppressWarnings("unchecked")
+  public static <T> Comparator<T> getComparator(TypeProtos.MinorType type) {
     switch (type) {
       case INTERVALDAY:
       case INTERVAL:
       case INTERVALYEAR:
-        return Comparator.nullsFirst(UnsignedBytes.lexicographicalComparator());
+        return (Comparator<T>) (Comparator<?>)
```

Comment:
```suggestion
        return (Comparator<?>)
```

Reply:
Added a comment about why this horrible hack is necessary. Basically, the method is poorly designed, but I didn't want to make major changes just before a release.

Reply:
Thanks for adding a comment with an explanation.